### PR TITLE
Chore(update): Enhance pod-autoscaler logs and update helper pod name

### PR DESCRIPTION
**Pull request by uditgaurav**
Signed-off-by: uditgaurav <udit@chaosnative.com>

#### This PR is for:

- Enhancing the pod autoscaler experiment logging.
- Add helper keyword in the helper pod name.
  - Replacer Available Replica to Ready Replica in the check.

### CODE REVIEW

**Comment by uditgaurav** [comment]
- Pod Autoscaler logs with failed case:

```bash
W0422 20:33:29.346756       1 client_config.go:541] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
time="2021-04-22T20:33:29Z" level=info msg="Experiment Name: pod-autoscaler"
time="2021-04-22T20:33:29Z" level=info msg="[PreReq]: Getting the ENV for the  experiment"
time="2021-04-22T20:33:29Z" level=info msg="[PreReq]: Updating the chaos result of pod-autoscaler experiment (SOT)"
time="2021-04-22T20:33:29Z" level=info msg="The application informations are as follows" Namespace=default AppKind=deployment AppLabel="app=nginx" Ramp Time=0
time="2021-04-22T20:33:29Z" level=info msg="[Status]: Verify that the AUT (Application Under Test) is running (pre-chaos)"
time="2021-04-22T20:33:29Z" level=info msg="[Status]: Checking whether application containers are in ready state"
time="2021-04-22T20:33:29Z" level=info msg="[Status]: The Container status are as follows" container=nginx Pod=nginx-86c57db685-2vzmt Readiness=true
time="2021-04-22T20:33:29Z" level=info msg="[Status]: The Container status are as follows" Pod=nginx-86c57db685-68h9z Readiness=true container=nginx
time="2021-04-22T20:33:29Z" level=info msg="[Status]: The Container status are as follows" container=nginx Pod=nginx-86c57db685-7j7tm Readiness=true
time="2021-04-22T20:33:29Z" level=info msg="[Status]: The Container status are as follows" container=nginx Pod=nginx-86c57db685-8rkgh Readiness=true
time="2021-04-22T20:33:29Z" level=info msg="[Status]: The Container status are as follows" Pod=nginx-86c57db685-b2d5p Readiness=true container=nginx
time="2021-04-22T20:33:29Z" level=info msg="[Status]: The Container status are as follows" Pod=nginx-86c57db685-htl9m Readiness=true container=nginx
time="2021-04-22T20:33:29Z" level=info msg="[Status]: The Container status are as follows" Readiness=true container=nginx Pod=nginx-86c57db685-j9hhl
time="2021-04-22T20:33:29Z" level=info msg="[Status]: The Container status are as follows" container=nginx Pod=nginx-86c57db685-r8rkb Readiness=true
time="2021-04-22T20:33:29Z" level=info msg="[Status]: The Container status are as follows" Readiness=true container=nginx Pod=nginx-86c57db685-wrwlx
time="2021-04-22T20:33:29Z" level=info msg="[Status]: The Container status are as follows" container=nginx Pod=nginx-86c57db685-ztrtx Readiness=true
time="2021-04-22T20:33:31Z" level=info msg="[Status]: Checking whether application pods are in running state"
time="2021-04-22T20:33:31Z" level=info msg="[Status]: The status of Pods are as follows" Pod=nginx-86c57db685-2vzmt Status=Running
time="2021-04-22T20:33:31Z" level=info msg="[Status]: The status of Pods are as follows" Pod=nginx-86c57db685-68h9z Status=Running
time="2021-04-22T20:33:31Z" level=info msg="[Status]: The status of Pods are as follows" Status=Running Pod=nginx-86c57db685-7j7tm
time="2021-04-22T20:33:31Z" level=info msg="[Status]: The status of Pods are as follows" Status=Running Pod=nginx-86c57db685-8rkgh
time="2021-04-22T20:33:31Z" level=info msg="[Status]: The status of Pods are as follows" Pod=nginx-86c57db685-b2d5p Status=Running
time="2021-04-22T20:33:31Z" level=info msg="[Status]: The status of Pods are as follows" Pod=nginx-86c57db685-htl9m Status=Running
time="2021-04-22T20:33:31Z" level=info msg="[Status]: The status of Pods are as follows" Status=Running Pod=nginx-86c57db685-j9hhl
time="2021-04-22T20:33:31Z" level=info msg="[Status]: The status of Pods are as follows" Pod=nginx-86c57db685-r8rkb Status=Running
time="2021-04-22T20:33:31Z" level=info msg="[Status]: The status of Pods are as follows" Pod=nginx-86c57db685-wrwlx Status=Running
time="2021-04-22T20:33:31Z" level=info msg="[Status]: The status of Pods are as follows" Pod=nginx-86c57db685-ztrtx Status=Running
time="2021-04-22T20:33:33Z" level=info msg="[Info]: Found deployment name 'nginx' with replica count '10'"
time="2021-04-22T20:33:33Z" level=info msg="[Info]: Details of Deployments under chaos injection" Number Of Deployment=1 Target Deployments="[nginx]"
time="2021-04-22T20:33:33Z" level=info msg="Updating deployment 'nginx' to number of replicas '100'"
time="2021-04-22T20:33:33Z" level=info msg="[Info]: The application started scaling"
time="2021-04-22T20:34:05Z" level=info msg="[Info]: Application started rolling back to original replica count"
time="2021-04-22T20:34:05Z" level=info msg="[Info]: Application ready replica count is: 49"
time="2021-04-22T20:34:07Z" level=info msg="[Info]: Application ready replica count is: 49"
time="2021-04-22T20:34:11Z" level=info msg="[RollBack]: Application rollback to the initial number of replicas"
time="2021-04-22T20:34:11Z" level=error msg="Chaos injection failed, err: fail to perform autoscaling, err: application deployment status check failed, err: fail to scale the deployment to the desired replica count in the given chaos duration"
```

File: chaoslib/litmus/node-cpu-hog/lib/node-cpu-hog.go [hunk 1]
```diff
@@ -116,13 +116,13 @@ func InjectChaosInSerialMode(experimentsDetails *experimentTypes.ExperimentDetai
...
 
-		appLabel := "name=" + experimentsDetails.ExperimentName + "-" + experimentsDetails.RunID
+		appLabel := "name=" + experimentsDetails.ExperimentName + "-helper-" + experimentsDetails.RunID
 
 		//Checking the status of helper pod
 		log.Info("[Status]: Checking the status of the helper pod")
 		err = status.CheckApplicationStatus(experimentsDetails.ChaosNamespace, appLabel, experimentsDetails.Timeout, experimentsDetails.Delay, clients)
 		if err != nil {
-			common.DeleteHelperPodBasedOnJobCleanupPolicy(experimentsDetails.ExperimentName+"-"+experimentsDetails.RunID, appLabel, chaosDetails, clients)
+			common.DeleteHelperPodBasedOnJobCleanupPolicy(experimentsDetails.ExperimentName+"-helper"+experimentsDetails.RunID, appLabel, chaosDetails, clients)
```
Comment (ispeakc0de) [inline chaoslib/litmus/node-cpu-hog/lib/node-cpu-hog.go:125]: hyphen missing

File: chaoslib/litmus/pod-autoscaler/lib/pod-autoscaler.go [hunk 2]
```diff
@@ -237,12 +237,11 @@ func DeploymentStatusCheck(experimentsDetails *experimentTypes.ExperimentDetails
...
 				if err != nil {
-					return errors.Errorf("Unable to find the deployment with name %v, err: %v", app.AppName, err)
+					return errors.Errorf("fail to find the deployment with name %v, err: %v", app.AppName, err)
 				}
-				log.Infof("Deployment's Available Replica Count is %v", deployment.Status.AvailableReplicas)
-				if int(deployment.Status.AvailableReplicas) != experimentsDetails.Replicas {
+				if int(deployment.Status.ReadyReplicas) != experimentsDetails.Replicas {
 					isFailed = true
-					return errors.Errorf("Application %s is not scaled yet, err: %v", app.AppName, err)
+					return errors.Errorf("application %s is not scaled yet, the desired replica count is: %v and redy replica count is: %v", app.AppName, experimentsDetails.Replicas, deployment.Status.ReadyReplicas)
```
Comment (ispeakc0de) [inline chaoslib/litmus/pod-autoscaler/lib/pod-autoscaler.go:244]:
```suggestion
					return errors.Errorf("application %s is not scaled yet, the desired replica count is: %v and ready replica count is: %v", app.AppName, experimentsDetails.Replicas, deployment.Status.ReadyReplicas)
```
```suggestion
					return errors.Errorf("application %s is not scaled yet, the desired replica count is: %v and redy replica count is: %v", app.AppName, experimentsDetails.Replicas, deployment.Status.ReadyReplicas)
```

File: chaoslib/litmus/kubelet-service-kill/lib/kubelet-service-kill.go [hunk 3]
```diff
@@ -71,13 +71,13 @@ func PrepareKubeletKill(experimentsDetails *experimentTypes.ExperimentDetails, c
...
 
-	appLabel := "name=" + experimentsDetails.ExperimentName + "-" + experimentsDetails.RunID
+	appLabel := "name=" + experimentsDetails.ExperimentName + "-helper-" + experimentsDetails.RunID
 
 	//Checking the status of helper pod
 	log.Info("[Status]: Checking the status of the helper pod")
 	err = status.CheckApplicationStatus(experimentsDetails.ChaosNamespace, appLabel, experimentsDetails.Timeout, experimentsDetails.Delay, clients)
 	if err != nil {
-		common.DeleteHelperPodBasedOnJobCleanupPolicy(experimentsDetails.ExperimentName+"-"+experimentsDetails.RunID, appLabel, chaosDetails, clients)
+		common.DeleteHelperPodBasedOnJobCleanupPolicy(experimentsDetails.ExperimentName+"-helper"+experimentsDetails.RunID, appLabel, chaosDetails, clients)
```
Comment (ispeakc0de) [inline chaoslib/litmus/kubelet-service-kill/lib/kubelet-service-kill.go:80]: hypen is missing, check other places as well

File: chaoslib/litmus/pod-autoscaler/lib/pod-autoscaler.go [hunk 4]
```diff
@@ -290,12 +289,11 @@ func StatefulsetStatusCheck(experimentsDetails *experimentTypes.ExperimentDetail
...
 				statefulset, err := appsv1StatefulsetClient.Get(app.AppName, metav1.GetOptions{})
 				if err != nil {
-					return errors.Errorf("Unable to find the statefulset with name %v, err: %v", app.AppName, err)
+					return errors.Errorf("fail to find the statefulset with name %v, err: %v", app.AppName, err)
 				}
-				log.Infof("Statefulset's Ready Replica Count is: %v", statefulset.Status.ReadyReplicas)
 				if int(statefulset.Status.ReadyReplicas) != experimentsDetails.Replicas {
 					isFailed = true
-					return errors.Errorf("Application is not scaled yet, err: %v", err)
+					return errors.Errorf("application %s is not scaled yet, the desired replica count is: %v and redy replica count is: %v", app.AppName, experimentsDetails.Replicas, statefulset.Status.ReadyReplicas)
```
Comment (ispeakc0de) [inline chaoslib/litmus/pod-autoscaler/lib/pod-autoscaler.go:296]:
```suggestion
					return errors.Errorf("application %s is not scaled yet, the desired replica count is: %v and ready replica count is: %v", app.AppName, experimentsDetails.Replicas, statefulset.Status.ReadyReplicas)
```
```suggestion
					return errors.Errorf("application %s is not scaled yet, the desired replica count is: %v and redy replica count is: %v", app.AppName, experimentsDetails.Replicas, statefulset.Status.ReadyReplicas)
```

File: chaoslib/litmus/pod-dns-chaos/lib/pod-dns-chaos.go [hunk 5]
```diff
@@ -114,7 +115,7 @@ func InjectChaosInSerialMode(experimentsDetails *experimentTypes.ExperimentDetai
 			return errors.Errorf("Unable to create the helper pod, err: %v", err)
 		}
 
-		appLabel := "name=" + experimentsDetails.ExperimentName + "-" + runID
+		appLabel := "name=" + experimentsDetails.ExperimentName + "-helper-" + runID
```
Comment (ispeakc0de) [inline chaoslib/litmus/pod-dns-chaos/lib/pod-dns-chaos.go:118]: Can we change other places as well?

Comment (uditgaurav) [reply chaoslib/litmus/pod-dns-chaos/lib/pod-dns-chaos.go:118]: Okay, what is the issue here?

**Review by ispeakc0de — APPROVED** [review]

**Review by ksatchit — APPROVED** [review]

**Pull request merged**